src/scripts/main.js

```diff
@@ -1,3 +1,192 @@
+/* eslint-disable max-len */
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tbody = document.querySelector('tbody');
+
+tbody.isSorted = false;
+tbody.currentIndex = 0;
+
+function tableSort(indexCol = tbody.currentIndex) {
+  const rowsForSort = [...tbody.querySelectorAll('tr')];
+
+  rowsForSort.sort((a, b) => {
+    const first = a.children[indexCol].innerText;
+    const second = b.children[indexCol].innerText;
+
+    if (isNaN(parseFloat(first.slice(1)))) {
+      return first.localeCompare(second);
+    } else {
+      if (first.slice(0, 1) === '$') {
+        return parseFloat(first.slice(1)) - parseFloat(second.slice(1));
+      }
+
+      return parseFloat(first) - parseFloat(second);
+    }
+  });
+
+  if (tbody.isSorted) {
+    rowsForSort.reverse().forEach((el) => tbody.append(el));
+    tbody.isSorted = false;
+  } else {
+    rowsForSort.forEach((el) => tbody.append(el));
+    tbody.isSorted = true;
+  }
+}
+
+table.addEventListener('click', (e) => {
+  if (e.target.tagName === 'TH') {
+    const place = e.target.closest('thead') ? 0 : table.rows.length - 1;
+
+    tbody.currentIndex = [...table.rows[place].children].indexOf(e.target);
+
+    tableSort(tbody.currentIndex);
+  }
+});
+
+tbody.addEventListener('click', (e) => {
+  const row = e.target.closest('TR');
+  const rowsForSort = [...tbody.querySelectorAll('tr')];
+
+  rowsForSort.forEach((item) => item.classList.remove('active'));
+  row.classList.add('active');
+});
+
+tbody.addEventListener('dblclick', (e) => {
+  if (e.target.tagName === 'TD') {
+    const cell = e.target;
+    const columnNum = [...cell.closest('TR').children].indexOf(cell);
+    const defaultValue = cell.innerText.trim();
+    const selection = window.getSelection();
+    const range = document.createRange();
+
+    cell.contentEditable = true;
+    range.selectNodeContents(cell);
+    selection.removeAllRanges();
+    selection.addRange(range);
+
+    cell.addEventListener('blur', () => {
+      const content = cell.innerText.trim();
+
+      if (columnNum === 4 && content.slice(0, 1) !== '$') {
+        if (!isNaN(parseFloat(content))) {
+          cell.innerText = '$' + (+content).toLocaleString('en');
```

Comment:
Use template literals for string concatenation in your code

Reply:
i don`t know what iis it...

Reply:
for example?